### PR TITLE
loki: query splitting: split logs queries

### PR DESCRIPTION
**Pull request by gabor**
this pull request adds the code to calculate split-time-ranges for logs-queries, and also allows single-logs-queries to be chunked.

how to test:
1. `make devenv sources=loki`
2. (wait a little, like 30 seconds, for data be inserted into loki..  now we have logs for the last 7 days)
3. open the network tab and be ready to look at the ajax requests
4. adjust the grafana-time-range to last-30-days
5. run a loki query that finds something in the oldest chunks, for example: `{place="luna"} |= "counter=5 "` (note the space at the end)
6. you should see that a lot of queries are run (one for each day), and then the requested log-line is found and displayed
7. (NOTE: if enough log-lines are found, we stop asking for more data,that was implemented in a previous PR)

### CODE REVIEW

File: public/app/plugins/datasource/loki/logsTimeSplit.ts [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+// - 100<>120
+// - 120<>140
+// - 140<>150
+// and no log-line will be skipped or duplicated
+// (NOTE: we do these calculations in milliseconds. at the end, Loki receives
+// nanoseconds, but it will be OK, because it's simply a matter to adding `000000`,
+// to the end, so if we do it right in milliseconds, it should be OK in
+// nanoseconds too
+
+export function getRanges(
```
Comment (matyax) [inline public/app/plugins/datasource/loki/logsTimeSplit.ts:21]: Really minor, but could you rename both `getRanges` functions to something more related with its function. E.g. splitTimeRenge or getRangeChunks.

**Review by matyax — COMMENTED** [review]
Looks good!

File: public/app/plugins/datasource/loki/logsTimeSplit.ts (hunk 1 above)
Comment (gabor) [reply public/app/plugins/datasource/loki/logsTimeSplit.ts]: sure, done!

File: public/app/plugins/datasource/loki/queryUtils.ts [hunk 2]
```diff
@@ -312,10 +312,6 @@ export function requestSupportsPartitioning(queries: LokiQuery[]) {
     return false;
   }
 
-  if (isLogsQuery(queries[0].expr)) {
```
Comment (matyax) [inline public/app/plugins/datasource/loki/queryUtils.ts:315]: Just double checking that with this PR you will be enabling logs chunking. If you meant to make this change, 👍

**Review by matyax — APPROVED** [review]
Great work!

File: public/app/plugins/datasource/loki/queryUtils.ts (hunk 2 above)
Comment (gabor) [reply public/app/plugins/datasource/loki/queryUtils.ts:315]: yes, thanks for checking, it is intentional. 
i understand the logs-visualization is not perfect yet regarding this, but it is good enough for a first iteration based on my tests (i mean, it does not crash 😄 ).

**Pull request merged**